### PR TITLE
web: move panel components into blob page

### PR DESCRIPTION
**Pull request by valerybugakov**
## Context

Based on the @philipp-spiess [work](https://github.com/sourcegraph/sourcegraph/pull/47904) with fixes related to CSS regressions. I created a separate branch to preserve commit history from the original PR with visual regressions and avoid resolving merge conflicts.

## Test plan

1. CI + local testing
2. [Main-dry-run](https://buildkite.com/sourcegraph/sourcegraph/builds/206282)

## App preview:

- [Web](https://sg-web-vb-ps-move-ref-panel-into-blob.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by valerybugakov** [comment]
Status: WIP (the PR is not in draft to trigger all visual regression checks on CI)

File: client/web/src/LegacyLayout.tsx [hunk 1]
```diff
@@ -226,22 +223,7 @@ export const LegacyLayout: FC<LegacyLayoutProps> = props => {
...
-                    id="references-panel"
-                >
-                    <TabbedPanelContent
-                        {...props}
-                        repoName={`git://${parseBrowserRepoURL(location.pathname).repoName}`}
-                        fetchHighlightedFileLineRanges={props.fetchHighlightedFileLineRanges}
-                    />
-                </Panel>
-            )}
+            <div id="references-panel-react-portal" />
```
Comment (valerybugakov) [inline client/web/src/LegacyLayout.tsx:226]: Going to find a repro on why this element needs to be inside of the suspense boundary and move + add a comment after that.

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.module.scss [hunk 2]
```diff
@@ -108,28 +108,3 @@
 .step-output-container {
     background-color: var(--gray-10);
 }
-
-.step-tabs {
```
Comment (valerybugakov) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.module.scss:112]: All these styles are default to the Wildcard `Tabs` component

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss [hunk 3]
```diff
@@ -1,27 +0,0 @@
-.batch-change-tabs {
-    [data-reach-tab],
-    [data-reach-tab]:hover,
-    [data-reach-tab][data-selected] {
-        padding: 0.5rem 1rem;
```
Comment (valerybugakov) [inline client/web/src/enterprise/batches/BatchChangeTabs.module.scss:5]: Use default padding to be consistent with other instances of the `Tabs` component.

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx [hunk 4]
```diff
@@ -600,8 +600,7 @@ const WorkspaceStep: React.FunctionComponent<React.PropsWithChildren<WorkspaceSt
                     <CardBody>
                         {!step.skipped && (
                             <Tabs
-                                className={styles.stepTabs}
-                                size="small"
+                                size="medium"
```
Comment (eseliger) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx:603]: is the change from small to medium intended?

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss [hunk 5]
```diff
@@ -1,27 +0,0 @@
-.batch-change-tabs {
```
Comment (eseliger) [inline client/web/src/enterprise/batches/BatchChangeTabs.module.scss:1]: Is this change related to this PR?

**Review by eseliger — APPROVED** [review]

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx (hunk 4 above)
Comment (valerybugakov) [reply client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx:603]: yes, it preserves the font-size of the current version

**Review by philipp-spiess — APPROVED** [review]

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss (hunk 5 above)
Comment (valerybugakov) [reply client/web/src/enterprise/batches/BatchChangeTabs.module.scss:1]: Yes, all these styles are redundant because they are specified in the wildcard `Tabs` component. Since I removed custom padding, I also removed all the other styles because they make no difference.

File: client/web/src/LegacyLayout.tsx (hunk 1 above)
Comment (valerybugakov) [reply client/web/src/LegacyLayout.tsx]: @philipp-spiess, good thing that you found this behavior! Here's the update: https://github.com/sourcegraph/sourcegraph/pull/49190/commits/cce593efb187958a4712e773cbaa560c5082a7b8

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss (hunk 5 above)
Comment (vovakulikov) [reply client/web/src/enterprise/batches/BatchChangeTabs.module.scss:1]: > I think this PR caused a regression in batch changes UIs in that it now causes layout shifts when tabs are activated. The previous styles made sure that we didn't get wider when the text became bold.

I think this part we should fix this in the tab component implementation (I have an idea about a generic solution without hardcoded values and modes.
